va/validation-authority.go

```diff
@@ -212,12 +208,10 @@ func (va *ValidationAuthorityImpl) fetchHTTP(identifier core.AcmeIdentifier, pat
 		scheme = "https"
 		port = va.httpsPort
 	}
-	portString := strconv.Itoa(port)
-	hostPort := net.JoinHostPort(host, portString)
 
 	url := &url.URL{
 		Scheme: scheme,
-		Host:   hostPort,
+		Host:   host,
```

Comment:
Correct me if I'm wrong: I think when port is not the default port for a given protocol, it is actually important to include it in the HTTP Host header, or a virtualhosting web server may misdirect the request. And I think this code always leaves off the port, right? I'd prefer something that includes the port when it is non-default (i.e. not 80 or 443).